Tutorial 2: Integers and Floating Points

The previous tutorial was focused exclusively on strings. In this tutorial, we’ll take a look at two different variable types: int (short for integer) and float (short for floating-point decimal)...negative whole numbers (numbers without decimal points).

In [1]:
HR = 2
print ('HR:',HR)
print ('type(HR)', type(HR))

HR: 2
type(HR) <class 'int'>


The first line defines the variable HR (Home Runs) as 2. The print statement on the second line outputs HR, and the third line has a print statement with a type function. The type function will tell us the class or type of the variable. In the output window, we have HR (2), and below it, the type of HR, which is int. 

In [2]:
singles = 5
HR = 2
team_hits = singles + HR
print ('team_hits:',team_hits)

team_hits: 7


Added to HR (2) is the variable singles (5), which is also an int. The third line of code has the formula for team_hits. The print statement shows that team_hits equals 7. See if you can add the variables doubles (value=2) and triples (value=1) to the formula and make team_hits equal 10. 

In [3]:
singles = 5
doubles = 2
triples = 1
HR = 2
team_hits = singles + doubles + triples + HR
print ('team_hits:',team_hits)

team_hits: 10


You might be tempted to name your variables “2B” and “3B”, but if you recall from Tutorial 1, variable names cannot begin with a number. Below is another formula with integers, this time getting the At Bats (AB) from Plate Appearances (PA).  

In [4]:
PA = 100
BB = 6
HBP = 1
sac_fly = 1
AB = PA - (BB + HBP + sac_fly)
print ('AB:', AB)

AB: 92


Next, we’ll take a look at some variables that represent floating-point numbers. Instead of type int, they will be of type float.

In [5]:
BA = 0.301
print ('BA:',BA)
print ('type(BA)', type(BA))

BA: 0.301
type(BA) <class 'float'>


The first line sets variable BA (Batting Average) to 0.301. The print statement shows us that BA is equal to 0.301, which includes both a zero and a decimal point. The zero followed by a decimal point is a clue that BA might be a float, and the type function confirms BA is indeed <class 'float'>.

In [6]:
OBP = 0.353
SLG = 0.549
OPS = OBP + SLG
print ('OPS:',OPS)

OPS: 0.902


The first two lines of code establish OBP (On-Base Percentage) and SLG (Slugging Percentage). Both variables are floats. The formula OPS is the sum of OBP and SLG. Next, we’ll take a look at a formula that combines a float with an int. 

In [7]:
TB = 101
AB = 201.0
slg_pct = TB/AB
print ('slg_pct:',slg_pct)

slg_pct: 0.5024875621890548


The variables TB (Total Bases) is an int at 101, and AB (At-Bats) is a float at 201.0. The third line is a formula that includes the division operator (/). Specifically, TB (an int) is being divided by AB (a float). This brings us to the output, which has slg_pct equal to 0.502, plus another 13 digits. Those extra digits are difficult to read, so let’s round it down with a built-in Python function called: round(). 

In [8]:
slg_pct = 0.5024875621890548
round_slg_pct = round(slg_pct, 3)
print ('round_slg_pct:',round_slg_pct)

round_slg_pct: 0.502


Much better! The round() function takes two arguments: first, the variable or number to be rounded, second, the number of places from the decimal point. In this case, round_slg_pct was rounded 3 places to 0.503. Next, we’ll look at the ERA formula, which includes the multiplication operator, the asterisk symbol (*). 

In [9]:
IP = 180.0
ER = 60
ERA = (ER * 9.0)/IP
print ('ERA:',round(ERA, 2))

ERA: 3.0


The ERA formula has 9.0 being multiplied by ER (Earned Runs). In the formula itself, (ER * 9.0) is in parentheses, so this term will be calculated first. The product is then divided by IP (Innings Pitched).

The result >>ERA: 3.0 is a nice even number, but that raises an interesting question: is the ERA variable a float or an int? Well, one clue is that it has a decimal point followed by a zero. That suggests ERA (the formula, but also the variable) is a float.

Coding Tip. In Python, there’s no difference between a variable and a formula. Indeed, there is no set classification in Python for formula. When a formula contains both an int and a float, the formula itself will become a float. In fact, if one of the variables is a float, the formula or variable will always be type float.

In [10]:
fav_pitcher = ''
IP, ER = 180.0, 60
ERA = (ER * 9.0)/IP
print ('type(fav_pitcher) ', type( fav_pitcher))
print ('type(ER) ', type(ER))
print ('type(ERA) ', type(ERA))

type(fav_pitcher)  <class 'str'>
type(ER)  <class 'int'>
type(ERA)  <class 'float'>


This code shows three different types of variables: str (string), int, and float. The variable fav_player is an empty string, IP is an integer, and ERA is a float. The type function tells us the type or class of the variable in the print statements.

Next, we’ll take a look at how a variable of one type can be converted to another type.

In [11]:
PA = 100
PA_as_float = float(PA)
print ('PA:',PA,' ... PA_as_float:',PA_as_float)

PA: 100  ... PA_as_float: 100.0


In the print statement, PA is an integer at 100. The variable PA_as_float is a float at 100.0. Once again, the decimal point and the zero are a clue. Below, we look at how a float can be changed back to an integer. 

In [12]:
ERA = 4.00
ERA_as_int = int(ERA)
print ('ERA:',ERA,' ... ERA_as_int:',ERA_as_int)

ERA: 4.0  ... ERA_as_int: 4


But watch out—making an ERA of 4.99 into int gives you 4, not 5. Don’t believe me? Try it for yourself. 

Next, we have a number (ER = '80'), that is not, technically speaking, a number; it’s a string. 


In [13]:
IP = 200
ER = '80'
print ('ER:',ER,' ... type(ER)',type(ER))

ER: 80  ... type(ER) <class 'str'>


It’s a little deceiving because when ER prints out, it shows 80, which looks like an int. But by running type(ER), we can see it’s a string. To get the formula to run, we need to use either int() or float(). 

In [14]:
IP = 200
ER = '80'
ERA = (float(ER) * 9.0)/IP
print ('ERA:',ERA)

ERA: 3.6


By the way, a string will only convert into a number when all characters are numbers (0–9), or in the case of a float, when a decimal is present. For instance, these three elements: '$100', '72F', and '~3.14' couldn’t be converted into a float or an int without additional work. See if you can turn those “dirty” strings into floats. 

In [15]:
#'$100', '72F', '~3.14'
d = '$100'
split_d = d.split('$')[1]
print ('split_d:',float( split_d) + 0.7)
################ 72F ###################
f = '72F'
slice_f = f[0:2]
print ('slice_f:',float(slice_f) + 8.8)
################ ~3.14 ################
pi = '~3.14'
slice_pi = pi[1:]
print ('slice_pi:',float(slice_pi) + 0.00159265359 )

split_d: 100.7
slice_f: 80.8
slice_pi: 3.14159265359


That about does it for this tutorial on ints and floats. As you get more familiar with Python, you’ll learn when to use ints and when to use floats. As for my own workflow, whenever I’m dealing with a formula, I mostly use floats. But there are many scenarios in Python where floats just won’t work. In fact, we encountered one scenario in the last chapter, where we looked at the index of a character in a string; that index was an int. Similarly, the next chapter on arrays (or lists) will also use a lot of ints. 